caffe2/core/typeid.cc

```diff
@@ -32,7 +35,7 @@ string Demangle(const char* name) {
   int status = 0;
   auto demangled = ::abi::__cxa_demangle(name, nullptr, nullptr, &status);
   if (demangled) {
-    auto guard = MakeGuard([demangled]() { free(demangled); });
+    auto guard = caffe2::MakeGuard([demangled]() { free(demangled); });
```

Comment:
Move MakeGuard to c10 namespace?

Reply:
Let's try to minimize movements at this stage as they're painful.

How about we just bring a bunch of necessary `using caffe2::MakeGuard` and have clean up process later? Or even just use caffe2 namespace directly for the parts which mostly get inherited from there (e.g. `using caffe2::TypeMeta`).